### ISSUE TITLE
[docs] (meta) Incrementally improve top 25 learning workflows with baseline evaluations

### ISSUE THREAD

**Issue by prestonso** [gatsbyjs/gatsby#13708]
A meta issue to track incremental improvements to documentation captured in baseline evaluations of the [top 25 workflows](https://docs.google.com/spreadsheets/d/175iZyC8khLy1JQncvgyvAD_vXpfsHxswQcn6N7GDeRA/edit).

You can participate in this initiative too! See below for information about contributing.

# Summary

Thanks to @marcysutton and @jlengstorf in their [enumeration of the top 25 workflows](https://docs.google.com/spreadsheets/d/175iZyC8khLy1JQncvgyvAD_vXpfsHxswQcn6N7GDeRA/edit) that Gatsby developers seek out in documentation, we have a robust understanding of the most common needs for those using Gatsby on a day-to-day basis.

This meta issue is intended to provide a list of constituent issues targeting those top 25 workflows and the improvements based on those evaluations.

Contributions welcome! Please keep reading for evaluation criteria that we are applying to all of these learning workflows as well as an enumeration of the top 25 workflows and corresponding existing issues.

# Relevant information

## Evaluation criteria

There are six areas by which we are evaluating the top 25 learning workflows:

| Criterion | 😞 | 😐 | 😄 |
| --- | --- | --- | --- |
| **Searchability** | 5th page of results or nonexistent | 2nd–4th page of Google results | 1st page of Google results |
| **Discoverability** | Within 6+ clicks on .org (or trapped in GH issue) | Within 4-5 clicks on .org | Within 2-3 clicks on .org |
| **Completeness** | Entire procedures missing (6+ clicks required) | Some steps missing (4-5 clicks required) | Docs mostly or fully complete (no more than 2-3 clicks) |
| **Linkedness** | No links to other useful docs pages | Some links to other useful docs pages | Many links to other useful docs pages |
| **Tone and accessibility** (for tutorials and guides) | Negative or overbearing tone | Neutral tone | Friendly and helpful tone |
| **Tone and accessibility** (for docs and API pages) | Uninformative, list of prerequisites absent | Somewhat informative, prerequisites are somewhat incomplete | Informative, prerequisites are clear |
| **Style** | Many style issues (needs proofread) | Some style issues (e.g. capitalization) | Adheres fully to [style guide](https://www.gatsbyjs.org/contributing/gatsby-style-guide/) |

NB: All recommendations are prefixed with **[rec]** and collated later in each workflow evaluation.

## List of workflows

If an issue does not exist for a workflow (except where `[skip]` is indicated), create an issue for that workflow and link from it to this meta issue (see template below). Prefix the workflow below with the issue number. Boxes should only be checked once a pull request is merged in that addresses the improvements identified by the evaluation.

- [x] 1. #13710: Setting up a blog that pulls content from Markdown
- [x] 2. [skip]: Setting up Gatsby Preview with Contentful [gatsbyjs.com]
- [x] 3. #13712: Linking to Gatsby and non-Gatsby content (merged in #14036)
- [x] 4. #13715: Using starters (merged in #14036)
- [x] 5. #14529: Working with images and videos (partially addressed in #13170 and #14697, see also https://github.com/gatsbyjs/gatsby/issues/3346#issuecomment-483814732)
- [X] 6. #14846 14796: Finding a source plugin
- [x] 7. #13876: Installing and using WordPress
- [X] 8. #14635: Adding and organizing CSS (including Sass)
- [x] 9. #18242 Using Gatsby Themes 
- [X] 10. #14258: Embedding components in Markdown with MDX
- [X] 11. #16981 Working with GraphQL
- [X] 12. #18993 Building Apps with Gatsby
- [X] 13. #19768 Building for E-commerce
- [x] 14. #19966 Troubleshooting error messages from queries not working
- [ ] 15. Using Schema Customization
- [X] 16. #20232 Building Plugins and Themes
- [x] 17. #13804: Deploying to Netlify
- [x] 18.  #17703: Working with fonts and typography
- [x] 19. #20691: Deploying to other hosting services
- [x] 20. #11941: Adding an RSS feed
- [x] 21. #21278 Working with style libraries and components
- [ ] 22. Implementing search with Algolia
- [ ] 23. #17645: Implementing Google AMP
- [x] 24. Adding a contact form (Doc added: #14768)
- [ ] 25. npm vs. yarn

## Contributing

To contribute, create a new issue titled "[docs] [workflows] Name of workflow being evaluated" and copy the template below for the text. See #13710 for a reference example.

While conducting evaluations, ignore all existing knowledge of and expertise with Gatsby, and imagine yourself in the position of a novice Gatsby user.

<pre>

Part of the **Top 25 Learning Workflows initiative**. See #13708 for the meta issue that this issue falls under.

# User story

As a new Gatsby user, I want to [describe workflow as completely as possible].

# Evaluation

[Change emoji below based on your evaluation.]

| Search | Discover | Complete | Linked | Tone | Style | Overall |
| --- | --- | --- | --- | --- | --- | --- |
| 😐 | 😐 | 😐 | 😐 | 😐 | 😐 | 😐 |

# Steps taken to implement

[List out steps taken to implement the workflow, evaluating against each of the criteria in the process.]

# Recommendations

- [ ] [List recommendations during evaluation here to be picked up in a pull request. This issue may be closed when all recommendations are implemented or deprioritized.]

</pre>

## Acknowledgments

Thank you to @marcysutton, @shannonbux, and @marisamorby for their feedback during this process and to @marcysutton and @jlengstorf for the foundational work of identifying these learning workflows.

**Comment by prestonso** [comment]
Added `help wanted` label as this is a great way for new contributors to begin contributing to Gatsby documentation.

**Comment by prestonso** [comment]
Added a link to #13804 in the meta — thanks for your help, @aravindballa!

**Comment by shannonbux** [comment]
This looks excellent! I think part of the steps that will be crucial to
document are how people searched for docs related to each workflow.

**Comment by haezl** [comment]
`7. Installing and using WordPress` and `13. Installing and using Contentful` are probably ambiguous. Whats the scope of those workflows? Reading the title it seems unrelated to Gatsby but I think its about how to pull content from those CMS into Gatsby and create a blog/website?

**Comment by marcysutton** [comment]
@haezl everything on this list is in relation to Gatsby, so using headless CMS installs to pull in blog content. It is sort-of assumed because we're talking about Gatsby workflows

**Comment by haezl** [comment]
@marcysutton thanks for clarifying

Added #13876: Installing and using WordPress, looking forward to feedback.

**Comment by jasonellington** [comment]
> @marcysutton thanks for clarifying
> 
> Added #13876: Installing and using WordPress, looking forward to feedback.

I'll be adding an issue for **13. Installing and using Contentful** soon

**Comment by prestonso** [comment]
#13712 and #13715 are both now complete as of merged PR #14036!

**Comment by calcsam** [comment]
Nice! Note that #17 was addressed (completed?) with https://github.com/gatsbyjs/gatsby/pull/11941/ and #5 with https://github.com/gatsbyjs/gatsby/pull/13170

**Comment by jlengstorf** [comment]
@calcsam thanks! I edited the original issue to reflect that.

**Comment by prestonso** [comment]
@jlengstorf I don't think #13170 covers video handling — should that be separated out into another workflow?

**Comment by jlengstorf** [comment]
@prestonso great question — I think @marcysutton already did a bunch of work on the video workflows, though. Maybe she can add more info here?

**Comment by marcysutton** [comment]
Hey there. #13170 did add more docs for working with video, but it doesn't quite go all the way for HTML5 video. The page has a callout for requesting more details and there's an open issue:
- https://www.gatsbyjs.org/docs/working-with-video/#hosting-your-own-html5-video-files
- https://github.com/gatsbyjs/gatsby/issues/3346#issuecomment-483814732

**Comment by jlengstorf** [comment]
Thanks @marcysutton! I updated the issue to reflect that and unchecked the video workflow todo.

**Comment by gillkyle** [comment]
Just began work on:

10. Embedding components in Markdown with MDX

over in #14258.

**Comment by jlengstorf** [comment]
@gillkyle thanks! Feel free to edit this issue to add the reference to the checklist above!

**Comment by marcysutton** [comment]
I'm taking on the image workflow! Initial evaluation here: https://github.com/gatsbyjs/gatsby/issues/14529

**Comment by harshil1712** [comment]
Hey, I started to work for Using Gatsby Themes, and came across this #14107. Work going on for the same. I was wondering if I should go ahead with my work or should wait for the changes to take place. Thank you.

**Comment by harshil1712** [comment]
I would like to take 19. Implementing search with Algolia.

**Comment by brolag** [comment]
Taking Google AMP implementation https://github.com/gatsbyjs/gatsby/issues/17645

**Comment by jonniebigodes** [comment]
Go for it, can't wait to see it, been tracking the amp integration wirh Gatsby.

**Comment by leandromuto** [comment]
Is there anyone taking care of using Gatsby theme item? I would like to help with this one.

**Comment by marcysutton** [comment]
@leandromuto yes -- that one is being discussed right now internally. If anything changes though, I'll get in touch with you. Thanks for offering!

**Comment by nicholaspung** [comment]
Before I do anymore work, is "Working with fonts and typography" being worked on? Almost done with the evaluation criteria!

**Comment by leandromuto** [comment]
@marcysutton can you provide an updated list of the topics that are not being discussed? Thanks!

**Comment by nicholaspung** [comment]
If no one is taking "Working on fonts and typography", I've created an issue for it and would like to work on it! #17703

**Comment by marcysutton** [comment]
@leonardodino this is the most up-to-date list: I've updated it with some of the assignments and work that started recently. 

@nicholaspung sounds great! We've recently added some recipes for working with fonts, but I think there could be a whole [reference guide](https://www.gatsbyjs.org/contributing/docs-templates/#reference-guides) on it, in addition to the [Typography.js library doc](https://www.gatsbyjs.org/docs/typography-js/).

**Comment by leonardodino** [comment]
@marcysutton I think you meant @leandromuto 🙂

**Comment by LekoArts** [comment]
@bvlktech Please don't unpin the pinned issues. Thanks!

**Comment by bvlktech** [comment]
@LekoArts oh snap im sorry i thought that that was just on my end not everyones im sorry 🤦‍♀️ wont happen again my mistake

**Comment by thomaswangio** [comment]
> Is anyone taking care of using Gatsby theme item? I would like to help with this one.

Hi @leandromuto, I'm assigned to the "Using Themes" workflow! It's under issue #18242. I'll be working on it until at least 10/13. If you have any ideas regarding it, or want to make a PR related to the workflow, feel free to comment on the issue and we can tackle it together 😊

**Comment by gillkyle** [comment]
I'm going to begin working on #12 Building Apps with Gatsby soon, and will have an issue with a workflow assessment that links back to this issue when it's ready.

**Comment by sjku1** [comment]
Hi, I was thinking of taking on 20. Making reusable components. is it alright if I make an issue for it?

**Comment by kylefloros** [comment]
I created issue #19768 for
13. Building for E-commerce

**Comment by jamstack-elise** [comment]
I created issue #20691 for 19. Deploying to other hosting services 😊🖖

**Comment by mujeex** [comment]
Hey y'all!! Just created an issue #24641 for 22. Implementing Search with Algolia. Waiting for some feedback. Thanks.

PS: First contribution 😄

**Comment by ItzaMi** [comment]
Hi 👋 I just created the issue #25659 for npm vs. yarn. 
Thank you in advance for any feedback!

**Comment by pragmaticpat** [comment]
@meganesu and @LekoArts - I'm closing this one - it's not received meaningful interaction since July 2020, and our docs have evolved significantly since then.
